### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by DaveZag**
# Ruby on Rails capstone project - Budget app
### On this branch I implement the following:
- Splash screen
     A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
- Home page (categories page)
- Transactions page
- "Add a new category" page
- "Add a new transaction" page

- Technical requirements
  - [x] You should use Postgres as your database.
  - [x] You should use devise for authentication.
  - [x] You should validate a user input to make sure that anyone with bad intentions cannot compromise your app.
  - [x] You can use a view template engine of your choice (.erb, .slim, .haml).
  - [x] The project should be deployed and accessible online.
  - [x] Your database schema should reflect the following structure:   
       erd_diagram

[Link](https://drive.google.com/file/d/1XsLlwiT1LTqpRVhl1DDnkup10K6PLdiQ/view?usp=sharing) to the presentation video.
[Link](https://moneymate-5vc4.onrender.com/) to the app deployment.

### CODE REVIEW

File: app/views/groups/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    </div>
+  <%else%>
+    <div class='error-message flex justify-center p-2'>
+      <p class="shadow-lg bg-white p-3">No Transactions yet. Please add some.</p>
+    </div>
+  <%end%>
+
+   <%= button_to "Add Transaction", new_payment_path, method: :get, params: {current_group: @group.id}, class: "btn secondary-color transtaction-btn w-fit text-white ml-3 mt-2 md:text-lg md:self-center" %>
+
+</section>
```
Comment (thecodechaser) [inline app/views/groups/show.html.erb]: - [x] Please make sure at the top of the transaction page the user can see the total amount for the category (the sum of all of the amounts of transactions in that category) as requested in the project requirements. Currently, It's not displayed there on the page.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @DaveZag👋, 

### You are doing a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- All tests are passed. ✔️
- PR has a good title and summary. ✔️
- No linter errors. :heavy_check_mark:



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/show.html.erb (hunk 1 above)
Comment (DaveZag) [reply app/views/groups/show.html.erb]: Hi @thecodechaser, Thanks for the feedback.🙂

**Comment by DaveZag** [comment]
The requested changes have been implemented:
image

File: app/views/groups/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      </div>
+
+      <div class="select-icon flex flex-wrap">
+        <%= form.text_field :icon, class:"icon-input py-2 px-4 w-full border focus:outline-orange-400 md:w-4/6", placeholder: "Icon" %>
+        <p class="icons-bt btn primary-color mt-2 text-white">Select an Icon</p>
+      </div>
+
+      <div class="actions self-center">
+        <%= form.submit "Create", class:"btn secondary-color text-white px-8 py-2" %>
+      </div>
```
Comment (HAMMAS-SALEEM) [inline app/views/groups/new.html.erb:31]: ## Your Submit Button
image

## Design Submit Button
image

**[Optional]**

- [x] You can see that the submit button is at the bottom of the design but in your design, it is not at the bottom :fearful: Kindly fix this issue to make your project follow the design and professionalism :+1:
- [x] Secondly the button should have an inside white border which you can see in the design image above :arrow_up: Kindly fix it to make your project look attractive to your viewers and recruiters :+1:

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @DaveZag 👋

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Requirements
:heavy_check_mark: Following Gitflow

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/new.html.erb (hunk 2 above)
Comment (DaveZag) [reply app/views/groups/new.html.erb:31]: Hi @HAMMAS-SALEEM, thanks for the feedback. I'm going to implement those changes.

**Pull request merged**